### PR TITLE
[stable22] Background job time windows

### PR DESCRIPTION
**Pull request by nickvergessen**
Backport of https://github.com/nextcloud/server/pull/30945

### CODE REVIEW

**Comment by PVince81** [comment]
what bits are missing to finish this ?

also to not forget:
- [ ] port to stable23 (in case there were conflicts resolved here)

**Comment by nickvergessen** [comment]
> what bits are missing to finish this ?

- [x] Wait for master to be merged
- [ ] Make sure the Patch is still up to date in case master was modified
- [ ] Trigger backport bot from master for 23 (optional: close this and backport bot to 22 as well)

**Comment by nickvergessen** [comment]
Review please

**Review by PVince81 — APPROVED** [review]
:+1:

File: lib/public/BackgroundJob/IJobList.php [hunk 1]
```diff
@@ -85,10 +85,11 @@ public function getAll();
 	/**
 	 * get the next job in the list
 	 *
+	 * @param bool $onlyTimeSensitive
 	 * @return \OCP\BackgroundJob\IJob|null
-	 * @since 7.0.0
+	 * @since 7.0.0 - In 24.0.0 parameter $onlyTimeSensitive got added
```
Comment (blizzz) [inline lib/public/BackgroundJob/IJobList.php:90]: please adjust to the targeted 22 version

**Review by blizzz — APPROVED** [review]

**Pull request merged**